lang/c++/CMakeLists.txt

```diff
@@ -202,15 +196,16 @@ if (AVRO_BUILD_EXECUTABLES)
     gen (union_redundant_types redundant_types)
 
     add_executable (avrogencpp impl/avrogencpp.cc)
-    target_link_libraries (avrogencpp avrocpp_s ${Boost_LIBRARIES})
+    target_link_libraries (avrogencpp avrocpp_s)
+    target_compile_definitions(avrogencpp PRIVATE AVRO_VERSION="${AVRO_VERSION}")
 endif ()
 
 if (AVRO_BUILD_TESTS)
     enable_testing()
 
     macro (unittest name)
         add_executable (${name} test/${name}.cc)
-        target_link_libraries (${name} avrocpp_s ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES} ${ZLIB_LIBRARIES})
+        target_link_libraries (${name} avrocpp_s Boost::system ZLIB::ZLIB $<$<TARGET_EXISTS:Snappy::snappy>:Snappy::snappy>)
```

Comment:
Can't we use `$<TARGET_NAME_IF_EXISTS:Snappy::snappy>` here as well instead of `$<$<TARGET_EXISTS:Snappy::snappy>:Snappy::snappy>`?

Reply:
Yes, that's better. I have switched to use `TARGET_NAME_IF_EXISTS`. At the same time, the minimum CMake version has been bumped to 3.12 to use it:  https://cmake.org/cmake/help/v3.12/release/3.12.html?highlight=target_name_if_exists#generator-expressions